core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineStrategy.java

```diff
@@ -3,117 +3,76 @@
+
+  RdbEngine getRdbEngine();
+
+  String getDataTypeForEngine(DataType dataType);
+
+  String getDataTypeForKey(DataType dataType);
+
+  String getTextType(int charLength);
 
-  protected abstract String createTableInternalPrimaryKeyClause(
+  String computeBooleanValue(boolean value);
+
+  void createNamespaceExecute(Connection connection, String fullNamespace) throws SQLException;
+
+  String createTableInternalPrimaryKeyClause(
       boolean hasDescClusteringOrder, TableMetadata metadata);
 
-  protected abstract void createTableInternalExecuteAfterCreateTable(
+  void createTableInternalExecuteAfterCreateTable(
       boolean hasDescClusteringOrder,
       Connection connection,
       String schema,
       String table,
       TableMetadata metadata)
       throws SQLException;
 
-  abstract void createMetadataTableIfNotExistsExecute(
-      Connection connection, String createTableStatement) throws SQLException;
-
-  abstract void createMetadataSchemaIfNotExists(Connection connection, String metadataSchema)
+  void createMetadataTableIfNotExistsExecute(Connection connection, String createTableStatement)
       throws SQLException;
 
-  abstract void deleteMetadataSchema(Connection connection, String metadataSchema)
+  void createMetadataSchemaIfNotExists(Connection connection, String metadataSchema)
       throws SQLException;
 
-  abstract void dropNamespace(BasicDataSource dataSource, String namespace)
-      throws ExecutionException;
+  void deleteMetadataSchema(Connection connection, String metadataSchema) throws SQLException;
 
-  abstract String namespaceExistsStatement();
+  void dropNamespace(BasicDataSource dataSource, String namespace) throws ExecutionException;
 
-  abstract void alterColumnType(
+  String namespaceExistsStatement();
+
+  void alterColumnType(
       Connection connection, String namespace, String table, String columnName, String columnType)
       throws SQLException;
 
-  abstract void tableExistsInternalExecuteTableCheck(Connection connection, String fullTableName)
+  void tableExistsInternalExecuteTableCheck(Connection connection, String fullTableName)
       throws SQLException;
 
-  abstract void dropIndexExecute(
-      Connection connection, String schema, String table, String indexName) throws SQLException;
+  void dropIndexExecute(Connection connection, String schema, String table, String indexName)
+      throws SQLException;
 
-  protected String enclose(String name) {
-    return QueryUtils.enclose(name, getRdbEngine());
-  }
+  /**
+   * Enclose the target (schema, table or column) to use reserved words and special characters.
+   *
+   * @param name The target name to enclose
+   * @return An enclosed string of the target name
+   */
+  String enclose(String name);
 
-  protected String encloseFullTableName(String schema, String table) {
-    return enclose(schema) + "." + enclose(table);
-  }
+  String encloseFullTableName(String schema, String table);
```

Comment:
It looks like we can use a `default` method here like:
```suggestion
  default String encloseFullTableName(String schema, String table) {
    return enclose(schema) + "." + enclose(table);
  }
```

Reply:
Wow, that's what I want!

Reply:
421df1b9e1a8430874e7a716b82face68edd39b0 removes duplicate methods in implementations.